homeassistant/components/foscam/__init__.py

```diff
@@ -31,6 +60,12 @@ async def async_setup_entry(hass: HomeAssistant, entry: FoscamConfigEntry) -> bo
     )
 
     coordinator = FoscamCoordinator(hass, entry, session)
+    WEBHOOK_ID = async_generate_id()
+    webhook.async_register(hass, DOMAIN, entry.title, WEBHOOK_ID, handle_webhook)
+    webhook_url = async_generate_url(hass, WEBHOOK_ID)
+    await hass.async_add_executor_job(
+        coordinator.session.setAlarmHttpServer, webhook_url
+    )
```

Comment:
Okay so
1. we should persist the webhook id and reuse the current one
2. I think we should have the coordinator refresh first, as that one has proper error handling whenever we can't reach the device
3. What if the device can't reach HA? Maybe because of firewall.